ksmt-bitwuzla/src/main/kotlin/org/ksmt/solver/bitwuzla/KBitwuzlaSolver.kt

```diff
@@ -99,6 +99,7 @@ open class KBitwuzlaSolver(private val ctx: KContext) : KSolver<KBitwuzlaSolverC
         bitwuzlaCtx.bitwuzlaTry {
             ctx.ensureContextMatch(assumptions)
 
+            invalidatePreviousModel()
```

Comment:
Currently, we don't analyze assertions and assumptions and therefore we can't determine whether the solver state is the same.
 Also, `check-sat` with the same state can return a different model. Since models are evaluated lazily this will become  a problem.